cooc.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
```

Comment:
It's part of the manual verification suit of compliance with the previous implementation.
Replacing it with automated verification is part of the future work.